openadapt/app/tray.py

```diff
@@ -183,7 +225,125 @@ def _quit() -> None:
         # for storing toasts that should be manually removed
         self.sticky_toasts = {}
 
-        self.launch_dashboard()
+        # self.launch_dashboard()
+
+    def stop_download(self) -> None:
+        """Stops download when button clicked."""
+        self.cancel_download_event.set()
+        self.show_toast("Stopping download...", duration=4000)
+
+    @Slot(str)
+    def download_start_toast_slot(self, message: str) -> None:
+        """Shows download start toast depending on emitted signal."""
+        self.download_progress_toast = self.show_toast(
+            message,
+            duration=DOWNLOAD_TOAST_UPDATE_TIME * 1000 - 50,
```

Comment:
@shashank40 can you please clarify why `* 1000 - 50`? Ideally this would be clarified in code, e.g.

DOWNLOAD_TOAST_DURATION_MILLISECONDS = 3 * 1000 - 50  # subtract 50 for xyz reason

Reply:
Yess @abrichr , it was to handle condition between spitting new toast and old toast times out by then.
But i have changed it to .hide()